client/extensions/woocommerce/app/products/product-form-variations-card.js

```diff
@@ -22,7 +22,21 @@ const ProductFormVariationsCard = ( {
 } ) => {
 	const handleToggle = () => {
 		if ( 'variable' !== product.type ) {
-			editProduct( product, { type: 'variable', dimensions: {} } );
+			// Reset any 'simple' product settings that have been filled out,
```

Comment:
*shakes fist*. GitHub's UI didn't show me your second comment. Agreed. I think this PR is large enough so I can remove the toggle logic here, and be smarter about this in a future PR.